### PR TITLE
Adds SwapOf and SwapSwitch

### LINKED ISSUE #476: Add Swaps
### Expected Behavior
We have an object which represents conversion of input object(s) to an output object.
We also have a switch object to fork the swap based on a string condition.

### Actual Behavior
We have only func which leads to procedural thinking.

### The log given by the failure.
Interface

```csharp
/// <summary>
    /// Swaps input to output.
    /// </summary>
    public interface ISwap<TInput, TOutput>
    {
        /// <summary>
        /// Turn input to output.
        /// </summary>
        TOutput Flip(TInput input);
    }

    /// <summary>
    /// Turn two inputs to one output.
    /// </summary>
    public interface ISwap<TInput1, TInput2, TOutput>
    {
        /// <summary>
        /// Turn two inputs to one output.
        /// </summary>
        TOutput Flip(TInput1 input1, TInput2 input2);
    }

    /// <summary>
    /// Turn three inputs to one output.
    /// </summary>
    public interface ISwap<TInput1, TInput2, TInput3, TOutput>
    {
        /// <summary>
        /// Turn three inputs to one output.
        /// </summary>
        TOutput Flip(TInput1 input1, TInput2 input2, TInput3 input3);
    }
```

### LINKED ISSUE #528: We have an action that is chosen by given condition. --> ForkFunc
### Expected Behavior
We have ForkFunc from Yaapii.Atoms.Tmp 

### Actual Behavior
It is missing

### The log given by the failure.

### PR DESCRIPTION
**Pull request by jaanpeeter**
### Please check if the PR fulfills these requirements
- [x] I made sure that my code builds
- [x] I merged the master into this branch before pushing
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?** (check one with "x")
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

### What is the current behavior? (You can also link to an open issue here)

#528 
#476 
### What is the new behavior?

closes #528 
closes #476 
### Does this PR introduce a breaking change? (check one with "x")
- [ ] Yes
- [x] No

###  Other information

### CODE REVIEW

**Comment by koeeenig** [comment]
This behaviour might be better implemented with the interface proposed in #476 
AS mentioned in the issue, `Func` leads to procedural thinking. What do you think @MSE1188 @jaanpeeter  ?

**Comment by MSE1188** [comment]
Agree. ```ForkFunc``` is not a good name. ```Swap``` is better

**Comment by koeeenig** [comment]
I would add the `ISwap` interface and keep the `IFunc` interface as well. In my opinion it fits into the library because using `ISwap`  or `IFunc` depends on the usecase.

**Comment by Meerownymous** [comment]
From my perspective, func is something where I as the developer am the actor who triggers something that should happen with an input. It puts me in control and takes responsibility from the object. 

Swap can be seen as a permanent conversion, offering its services whenever needed.
I use it when I need a performant object which is seldomly constructed and does a lot of conversions during its lifetime. Because Atoms objects are all caching their result by default, the developer cannot use a chain of decorators for this - and making the these live comes at the cost of performance. 

To me swaps are handy to build a conversion object which is constructed as an app singleton.

Func and Swap are indeed identical and it is all about what story the developer intends their code to tell to future readers.

File: src/Yaapii.Atoms/Swap/SwapSwitch.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        {
+            return this.map[condition]
+                .Invoke();
+        }
+    }
+
+    /// <summary>
+    /// A tooth for a SwapSwitch
+    /// </summary>
+    public sealed class Tooth<TInput, TOutcome> : KvpEnvelope<TInput, Func<TOutcome>>
```
Comment (koeeenig) [inline src/Yaapii.Atoms/Swap/SwapSwitch.cs:53]: I suggest to rename `Tooth` to `SwapIf `..it's easier to understand.
Move the class to it's own file.

File: src/Yaapii.Atoms/Swap/SwapSwitch.cs
Comment (koeeenig) [inline src/Yaapii.Atoms/Swap/SwapSwitch.cs:1]: We had an implementation in the [PAQO](https://github.com/icarus-consulting/PAQO/blob/main/src/PAQO.Core/Facets/SwapSwitch.cs) library for `SwapSwitch` as well. Would you mind moving the ctors from there into this implementation?

**Review by koeeenig — CHANGES_REQUESTED** [review]

File: src/Yaapii.Atoms/Swap/SwapSwitch.cs
Comment (koeeenig) [reply src/Yaapii.Atoms/Swap/SwapSwitch.cs:1]: `SwapSwitch<TInput, TOutput> : ISwap<string, TInput, TOutput>` tights the switch condition to be based on a string which is in most cases sufficient. However someone might need another type as condition so i would suggest `SwapSwitch<TCondition, TInput, TOutput> : ISwap<TCondition, TInput, TOutput>`.

You can additionaly add a `SwapSwitch` which sets the TCondition to string as in `SwapSwitch<TInput, TOutput> : ISwap<string, TInput, TOutput>`.

**Review by MSE1188 — APPROVED** [review]

**Comment by MSE1188** [comment]
ISwap now replaces the IFunc interface. We should also remove IFunc and the FuncOf class to prevent redundancy or at least add ```Obsolete``` attributes to these classes

**Comment by Meerownymous** [comment]
I would not see Ifunc obsolete, as ISwap is a different thing: Where I expect IFunc to have effects beyond making an output when invoking, I would not expect ISawp to to anything else than give me the desired output. Hence it matters what the code is supposed to express.

I also think the original cactoos introduced it to allow for adopting non-interface library object "func" to interface and therefore allow the common eo patterns to be applied to it. The same like INumber or IText, which transform the primitives string and int/double etc to eo "types" (interfaces).

**Comment by koeeenig** [comment]
i agree with @Meerownymous , `IFunc` enables the EO principle to inject `Func` into a constructor. We should definitely keep that in the library.

**Review by koeeenig — APPROVED** [review]

**Pull request merged**